include/effects/strip/faneffects.h

```diff
@@ -936,6 +1005,45 @@ class FireFanEffect : public LEDStripEffect
     abHeat = std::make_unique<uint8_t[]>(CellCount());
   }
 
+  FireFanEffect(const JsonObjectConst& jsonObject)

```

Comment:
Is there any validation that the JSON object contains all the params, or what if it's missing a few, and they don't get initialized?

Reply:
There is no such validation at this time. In line with what I wrote earlier ("a serializable class is responsible for its own serialization"), only the serializable class can figure out what to do with invalid JSON. My reasoning was as follows:

- A serializable class (effect) can decide at deserialization time (construction) how to deal with missing/invalid properties.
- If an effect wants to prevent construction in case of certain scenario, its factory (i.e. its respective lambda in effectfactories,cpp) can perform the appropriate validation and decide to return nullptr instead of a new effect.

The reason I didn't implement any actual validation logic for individual effects is that it requires more knowledge of the logic of the effects than I currently possess.